torch/functional.py

```diff
@@ -102,12 +102,34 @@ def broadcast_shapes(*shapes):
     """
     # This wrapper exists to support variadic args.
     # TODO Movie this to C++ once the jit has better support for torch.Size.
-    with torch.no_grad():
-        scalar = torch.zeros((), device="cpu")
-        tensors = [scalar.expand(shape) for shape in shapes]
-        tensors = broadcast_tensors(*tensors)
-        return tensors[0].shape
-
+    max_len = 0
+    for shape in shapes:
+        if isinstance(shape, int):
+            if max_len < 1:
+                max_len = 1
+        elif isinstance(shape, tuple) or isinstance(shape, list):
+            s = len(shape)
+            if max_len < s:
+                max_len = s
+    result = [1] * max_len
+    for shape in shapes:
+        if isinstance(shape, int):
+            shape = (shape,)
+        if isinstance(shape, tuple) or isinstance(shape, list):
+            for i in range(-1, -1 - len(shape), -1):
+                if shape[i] < 0:
+                    raise RuntimeError("Trying to create tensor with negative dimension ({}): ({})"
+                                       .format(shape[i], shape[i]))
+                if not isinstance(shape[i], int):
```

Comment:
jit turns elements of `torch.Size` into tensors when tracing, so this check fails. You can just remove it, if elements of tuples are of the wrong type, `torch.Size(result)` will error out below.

Reply:
Got that. Thanks for the share. I've updated it :)